src/main/java/uk/gov/hmcts/ccd/domain/service/callbacks/retrycontext/CallbackRetryContextBuilder.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.hmcts.ccd.domain.service.callbacks.retrycontext;
+
+import com.google.common.collect.Lists;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.ApplicationParams;
+
+import java.util.List;
+
+@Service
+public class CallbackRetryContextBuilder {
+    public static final int CALLBACK_RETRY_INTERVAL_MULTIPLIER = 3;
+    private final List<Integer> defaultCallbackRetryIntervalsInSeconds;
+    private final Integer defaultCallbackTimeoutInSeconds;
+
+    public CallbackRetryContextBuilder(final ApplicationParams applicationParams) {
+        this.defaultCallbackRetryIntervalsInSeconds = applicationParams.getCallbackRetryIntervalsInSeconds();
+        this.defaultCallbackTimeoutInSeconds = applicationParams.getCallbackTimeoutInSeconds();
+    }
+
+    public List<CallbackRetryContext> buildCallbackRetryContexts(final List<Integer> callbackRetryTimeouts) {
+        List<CallbackRetryContext> retryContextList = Lists.newArrayList();
+        if (callbackRetryTimeouts.isEmpty() || isCallbackRetriesWithZeros(callbackRetryTimeouts)) {
```

Comment:
can you please add a comment to the ticket or update the description mentioning that if the callback config contains a 0 we ignore the config and use the defaults?

Reply:
Sure added to the ticket